python/tvm/contrib/hexagon/session.py

```diff
@@ -95,6 +94,25 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, exc_traceback):
         pass
 
+    @property
+    def device(self):
+        """Session device."""
+
+        if hasattr(self, "_device") and self._device is not None:
+            return self._device
+
+        if not hasattr(self, "_requires_cpu_device"):
+            assert (
+                False
+            ), "Device type is not set. 'set_device_type' should be called before accessing device."
```

Comment:
This error message instructs users to call `set_device_type`, which we may want to be an internal function.  The concern would be for future breakage, if code outside this class calls `hexagon_session.set_device_type`, we wouldn't be able to remove `set_device_type` without breaking that code.  I think there are some plans to remove the `kDLHexagon` type and instead use options within the `kDLCPU`, at which point we'd want to remove the device choice without breaking external usage.